crates/core/src/surfnet/svm.rs

```diff
@@ -937,91 +944,177 @@ impl SurfnetSvm {
     ///
     /// # Arguments
     /// * `slot` - The slot number to retrieve the block for.
+    /// * `config` - The configuration for the block retrieval.
     ///
     /// # Returns
     /// `Some(UiConfirmedBlock)` if found, or `None` if not present.
-    pub fn get_block_at_slot(&self, slot: Slot) -> Option<UiConfirmedBlock> {
-        // Retrieve block
+    pub fn get_block_at_slot(
+        &self,
+        slot: Slot,
+        config: &RpcBlockConfig,
+    ) -> Option<UiConfirmedBlock> {
         let block = self.blocks.get(&slot)?;
 
-        // Retrieve parent block
-
-        // Retrieve transactions
-        let mut transactions = vec![];
-        for signature in &block.signatures {
-            let Some(TransactionWithStatusMeta(_slot, tx, _meta, _err)) = self
-                .transactions
-                .get(signature)
-                .map(|t| t.expect_processed().clone())
-            else {
-                continue;
-            };
+        let num_reward_partitions = config.rewards.unwrap_or(false);
```

Comment:
This is a misleading variable name - it's not a number it's a bool. This should be used to decide if rewards are returned or not, not represent the number of reward partitions